### ISSUE TITLE
Something is severely different with intellisense and snippets

### ISSUE THREAD

**Issue by alexandrudima** [Microsoft/vscode#8758]
I believe snippets were not offered before when the line had content on it.

I expect `getTabFocusMode` to be selected when intellisense pops up.

![image](https://cloud.githubusercontent.com/assets/5047891/16584386/21b4f8b8-42bc-11e6-9c25-573174be06e4.png)

**Comment by alexandrudima** [comment]
`1.2.1` shows the same **bad** behaviour

**Comment by alexandrudima** [comment]
`1.1.1` also looks **bad**

**Comment by alexandrudima** [comment]
Works ok? in `0.10.6`

![image](https://cloud.githubusercontent.com/assets/5047891/16584690/949d90be-42bd-11e6-8bfe-6a58eba7a975.png)

**Comment by joaomoreno** [comment]
@jrieken @aeschli We gotta get a grip on this. What has changed and why?

**Comment by aeschli** [comment]
There hasn't been any changes recently, but that the strategy:
To avoid that snippets are too prominent in the intellisense proposals:
Offer a snippet only if the name matches the current word or if the completion position is after a whitespace/new line.

We had a more restricting policy for a short time where we would offer snippets only after a whitespace/new line, but had to change that again after user complaints.

The sort order changes were made by @jrieken.
IMO snippets must go out of code complete and we should have a separate key.

**Comment by joaomoreno** [comment]
> IMO snippets must go out of code complete and we should have a separate key.

All in for that. Do we have a third, @jrieken ?

**Comment by jrieken** [comment]
I agree but I know it's thin ice. Just add options to control snippet sorting. We have users that want snippets sorted to the top, we have users that want the opposite.

**Comment by emzero** [comment]
Related issue: https://github.com/Microsoft/vscode/issues/6667

We already know there are people that want snippets on top of everything and other people that want them at the bottom. As I suggested in the related issue, I'm pretty sure the snippets lovers would we just fine if the snippets are shown on top only if the cursor is on a empty line and outside function parenthesis and object literal brackets.

**Comment by aeschli** [comment]
duplicate of #5223

**Comment by alexandrudima** [comment]
@aeschli I'm sorry I do not agree this is a duplicate of #5223 - [sql] add support for mysql

**Comment by joaomoreno** [comment]
I believe he meant #9286
